src/core/shape_inference/include/scaled_dot_product_attention_shape_inference.hpp

```diff
@@ -88,9 +92,10 @@ std::vector<TRShape> shape_infer(const ScaledDotProductAttention* op,
     }
 
     if (has_scale_input) {
-        if (input_shapes[4].rank().is_static() && input_shapes[4].is_static()) {
-            const auto& scale_is_scalar = input_shapes[4].rank().compatible(0);
-            const auto& scale_has_one_elem = input_shapes[4].rank().compatible(1) && input_shapes[4][0].compatible(1);
+        const auto& scale_rank = input_shapes[4].rank();
+        if (scale_rank.is_static() || input_shapes[4].is_static()) {
+            const auto& scale_is_scalar = scale_rank.compatible(0);
+            const auto& scale_has_one_elem = scale_rank.compatible(1) && input_shapes[4][0].compatible(1);
```

Comment:
This is not exactly valid as compatible return true if rank is dynamic so accessing the dimension[0] is not valid then.

But as main check checks rank is static the rank length can be taken and checked if is 0 or 1 etc.